.github/workflows/lint.yaml

```diff
@@ -169,14 +169,8 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Set Rust version
-        run: echo "RUST_VERSION=$(make -s -C build.assets print-rust-version)" >> $GITHUB_ENV
-
       - name: Set up Rust
         run: |
-          echo "Setting up Rust version ${RUST_VERSION}"
-          rustup toolchain install ${RUST_VERSION} --component rustfmt,clippy
-          rustup override set ${RUST_VERSION}
```

Comment:
I think we want a specific step for `rustup install` to separate the download and installation of the appropriate components from the actual builds.